applications/DeepSpeed-Chat/training/utils/perf.py

```diff
@@ -48,79 +41,126 @@ def print_throughput(hf_model, args, e2e_time, rank=0):
+                (actor_vocab_size /
+                 (16.0 * actor_num_layers * actor_hidden_size)))
 
         gen_tflops = gen_flops_per_iteration / (gen_exp_time * gpus_per_model *
                                                 (10**12))
 
-        if hf_config.torch_dtype == "float16":
+        if actor_hf_config.torch_dtype == torch.float16:
             num_bytes = 2
-        elif hf_config.torch_dtype == "float32":
+        elif actor_hf_config.torch_dtype == torch.float32:
             num_bytes = 4
         else:
-            num_bytes = 1
+            num_bytes = -1
 
-        gen_bw = (hf_model._num_params *
-                  (num_bytes / 1e9)) / gen_exp_time * args.max_answer_seq_len
+        pertok_lat = gen_exp_time / args.max_answer_seq_len
+        gen_bw = 1 / pertok_lat * actor_model._num_params * num_bytes / 1e9
 
-        total_flops_per_iteration = train_flops_per_iteration + gen_flops_per_iteration * args.generation_batches
+        total_flops_per_iteration = total_train_flops + gen_flops_per_iteration * args.generation_batches
         total_tflops = total_flops_per_iteration / (e2e_time * gpus_per_model *
                                                     (10**12))
 
         print(
-            f"End-to-End => Latency: {e2e_time:.2f}s, TFLOPs: {total_tflops:.2f}, Samples/sec: {samples_per_second:.2f}, Time/seq {e2e_time/batch_size:.2f}s, Batch Size: {batch_size}, Sequence Length: {seq_length}"
+            f"End-to-End => Latency: {e2e_time:.2f}s, TFLOPs: {total_tflops:.2f}, Samples/sec: {samples_per_second:.2f}, Time/seq {e2e_time/batch_size:.2f}s, Batch Size: {batch_size}, Total Seq. Length: {seq_length}"
         )
         print(
-            f"Generation => Latency: {gen_exp_time:.2f}s, TFLOPs: {gen_tflops:.2f}, BW: {gen_bw:.2f} GB/sec"
+            f"Generation => Latency: {gen_exp_time:.2f}s, Per-token Latency {pertok_lat*1000:.2f} ms, TFLOPs: {gen_tflops:.2f}, BW: {gen_bw if num_bytes > 0 else num_bytes:.2f} GB/sec, Answer Seq. Length: {args.max_answer_seq_len}"
         )
         print(
             f"Training   => Latency: {train_time:.2f}s, TFLOPs: {train_tflops:.2f}"
         )
-        param_string = f"{params_in_billions:.3f} B" if params_in_billions != 0 else "NA"
-        print(f"Parameters => {param_string}")
+        actor_param_string = f"{actor_params_in_billions:.3f} B" if actor_params_in_billions != 0 else "NA"
+        critic_param_string = f"{critic_params_in_billions:.3f} B" if critic_params_in_billions != 0 else "NA"
+        print(
+            f"Actor Model Parameters => {actor_param_string}, Critic Model Parameters => {critic_param_string}"
+        )
+
+
+# Helper function to calculate FLOPs using the Megatron-LM paper's formula
+def calculate_flops(checkpoint_activations_factor, batch_size, seq_length,
+                    hf_config):
+    num_layers, hidden_size, vocab_size = get_hf_configs(hf_config)
+    flops_per_iteration = (24 * checkpoint_activations_factor * batch_size *
```

Comment:
Nit: Can some of the "magic numbers" in the computation such as 24, 6.0, 16.0 change depending on the scenario?

Also, perhaps we can name the numbers e.g. `[name] = 24`?

Reply:
24 never changes and its based on the formula in megatron-lm paper so we are okay.